modules/tinymce/src/core/main/ts/api/Mode.ts

```diff
@@ -1,16 +1,38 @@
-import { Cell, Fun } from '@ephox/katamari';
+import { Cell, Fun, Obj, Type } from '@ephox/katamari';
 
 import { registerMode, setMode } from '../mode/Mode';
 import { isReadOnly, registerReadOnlyContentFilters, registerReadOnlySelectionBlockers } from '../mode/Readonly';
 import Editor from './Editor';
 
+export type ReadonlyProperty = Readonly<'uiEnabled' | 'selectionEnabled'>;
+
+export type EditorReadOnlyType = boolean | {
+  [K in ReadonlyProperty]?: boolean;
```

Comment:
Wouldn't it be easier to just list the two of them directly? We're not likely to have a lot of these 🤔